Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: d06b511a24853c73c10d7a13ee6479f2abc66de8
Number of Error(s): 4
/mnt/disk1/home/vamosax/smoketest/PR-12588/HPCC-Platform/plugins/cassandra/cassandraembed.cpp:280:84: error: \xe2\x80\x98cass_cluster_set_consistency\xe2\x80\x99 was not declared in this scope
make[2]: *** [plugins/cassandra/CMakeFiles/cassandraembed.dir/cassandraembed.cpp.o] Error 1
make[1]: *** [plugins/cassandra/CMakeFiles/cassandraembed.dir/all] Error 2
make: *** [all] Error 2

[Build error:](https://gist.github.com/cf8459c7bcace8212794a9e925931399)
Build log [Build log](https://gist.github.com/HPCCSmoketest/cf8459c7bcace8212794a9e925931399#file-result-19-06-06-10-21-43-log)

--- a/esp/src/eclwatch/WUQueryWidget.js
+++ b/esp/src/eclwatch/WUQueryWidget.js
@@ -257,6 +257,11 @@ define([
                     retVal.StartDate = date.add(now, "day", retVal.LastNDays * -1).toISOString();
                     retVal.EndDate = now.toISOString();
                 }
+                if (retVal.Type === "archived workunits") {
+                    lang.mixin(retVal, {
+                        timeOutSeconds: 300
+                    });
+                }
                 return retVal;
             },
 

--- a/esp/src/src/ESPRequest.ts
+++ b/esp/src/src/ESPRequest.ts
@@ -16,9 +16,11 @@ declare const debugConfig: any;
 class RequestHelper {
 
     serverIP: null;
+    timeOutSeconds: number;
 
     constructor() {
         this.serverIP = (typeof debugConfig !== "undefined") ? debugConfig.IP : this.getParamFromURL("ServerIP");
+        this.timeOutSeconds = 60;
     }
 
     getParamFromURL(key) {
@@ -102,10 +104,10 @@ class RequestHelper {
 
         var retVal = null;
         if (this.isCrossSite()) {
-            var transport = new hpccComms.Connection({ baseUrl: this.getBaseURL(service), type: "jsonp" });
+            var transport = new hpccComms.Connection({ baseUrl: this.getBaseURL(service), timeoutSecs: params.request.timeOutSeconds || this.timeOutSeconds, type: "jsonp" });
             retVal = transport.send(action + postfix, params.request, handleAs === "text" ? "text" : "json");
         } else {
-            var transport = new hpccComms.Connection({ baseUrl: this.getBaseURL(service) });
+            var transport = new hpccComms.Connection({ baseUrl: this.getBaseURL(service), timeoutSecs: params.request.timeOutSeconds || this.timeOutSeconds});
             retVal = transport.send(action + postfix, params.request, handleAs === "text" ? "text" : "json");
         }
 

